Written by ponce@slackbuilds.org trying to build and fixing wherever necessary

diff -Naur SFML-1.6.orig/include/SFML/System/ResourcePtr.inl SFML-1.6/include/SFML/System/ResourcePtr.inl
--- SFML-1.6.orig/include/SFML/System/ResourcePtr.inl	2010-01-27 14:53:54.000000000 +0100
+++ SFML-1.6/include/SFML/System/ResourcePtr.inl	2012-09-23 19:19:33.736300968 +0200
@@ -22,6 +22,7 @@
 //
 ////////////////////////////////////////////////////////////
 
+#include <unistd.h>
 
 ////////////////////////////////////////////////////////////
 /// Default constructor
diff -Naur SFML-1.6.orig/src/SFML/Audio/Listener.cpp SFML-1.6/src/SFML/Audio/Listener.cpp
--- SFML-1.6.orig/src/SFML/Audio/Listener.cpp	2010-01-27 15:00:19.000000000 +0100
+++ SFML-1.6/src/SFML/Audio/Listener.cpp	2012-09-23 19:23:25.645300966 +0200
@@ -25,6 +25,8 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
+#include <unistd.h>
+#include <stddef.h>
 #include <SFML/Audio/Listener.hpp>
 #include <SFML/Audio/OpenAL.hpp>
 
diff -Naur SFML-1.6.orig/src/SFML/Audio/Music.cpp SFML-1.6/src/SFML/Audio/Music.cpp
--- SFML-1.6.orig/src/SFML/Audio/Music.cpp	2010-01-27 15:00:19.000000000 +0100
+++ SFML-1.6/src/SFML/Audio/Music.cpp	2012-09-23 19:33:23.931300967 +0200
@@ -25,6 +25,8 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
+#include <unistd.h>
+#include <stddef.h>
 #include <SFML/Audio/Music.hpp>
 #include <SFML/Audio/OpenAL.hpp>
 #include <SFML/Audio/SoundFile.hpp>
diff -Naur SFML-1.6.orig/src/SFML/Audio/Sound.cpp SFML-1.6/src/SFML/Audio/Sound.cpp
--- SFML-1.6.orig/src/SFML/Audio/Sound.cpp	2010-01-27 15:00:19.000000000 +0100
+++ SFML-1.6/src/SFML/Audio/Sound.cpp	2012-09-23 19:26:57.001300967 +0200
@@ -25,6 +25,8 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
+#include <unistd.h>
+#include <stddef.h>
 #include <SFML/Audio/Sound.hpp>
 #include <SFML/Audio/SoundBuffer.hpp>
 #include <SFML/Audio/OpenAL.hpp>
diff -Naur SFML-1.6.orig/src/SFML/Audio/SoundBuffer.cpp SFML-1.6/src/SFML/Audio/SoundBuffer.cpp
--- SFML-1.6.orig/src/SFML/Audio/SoundBuffer.cpp	2010-01-27 15:00:19.000000000 +0100
+++ SFML-1.6/src/SFML/Audio/SoundBuffer.cpp	2012-09-23 19:29:59.239300967 +0200
@@ -25,6 +25,8 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
+#include <unistd.h>
+#include <stddef.h>
 #include <SFML/Audio/SoundBuffer.hpp>
 #include <SFML/Audio/SoundFile.hpp>
 #include <SFML/Audio/Sound.hpp>
diff -Naur SFML-1.6.orig/src/SFML/Audio/SoundBufferRecorder.cpp SFML-1.6/src/SFML/Audio/SoundBufferRecorder.cpp
--- SFML-1.6.orig/src/SFML/Audio/SoundBufferRecorder.cpp	2010-01-27 15:00:19.000000000 +0100
+++ SFML-1.6/src/SFML/Audio/SoundBufferRecorder.cpp	2012-09-23 19:26:17.856300966 +0200
@@ -25,6 +25,8 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
+#include <unistd.h>
+#include <stddef.h>
 #include <SFML/Audio/SoundBufferRecorder.hpp>
 #include <algorithm>
 #include <iterator>
diff -Naur SFML-1.6.orig/src/SFML/Audio/SoundStream.cpp SFML-1.6/src/SFML/Audio/SoundStream.cpp
--- SFML-1.6.orig/src/SFML/Audio/SoundStream.cpp	2010-03-18 14:14:46.000000000 +0100
+++ SFML-1.6/src/SFML/Audio/SoundStream.cpp	2012-09-23 19:34:43.477300968 +0200
@@ -25,6 +25,8 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
+#include <unistd.h>
+#include <stddef.h>
 #include <SFML/Audio/SoundStream.hpp>
 #include <SFML/Audio/AudioDevice.hpp>
 #include <SFML/Audio/OpenAL.hpp>
diff -Naur SFML-1.6.orig/src/SFML/Graphics/Font.cpp SFML-1.6/src/SFML/Graphics/Font.cpp
--- SFML-1.6.orig/src/SFML/Graphics/Font.cpp	2010-03-18 14:14:41.000000000 +0100
+++ SFML-1.6/src/SFML/Graphics/Font.cpp	2012-09-23 19:32:06.580300966 +0200
@@ -25,6 +25,8 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
+#include <unistd.h>
+#include <stddef.h>
 #include <SFML/Graphics/Font.hpp>
 #include <SFML/Graphics/FontLoader.hpp>
 #include <iostream>
diff -Naur SFML-1.6.orig/src/SFML/Graphics/Image.cpp SFML-1.6/src/SFML/Graphics/Image.cpp
--- SFML-1.6.orig/src/SFML/Graphics/Image.cpp	2010-01-27 15:00:05.000000000 +0100
+++ SFML-1.6/src/SFML/Graphics/Image.cpp	2012-09-23 19:22:22.316300968 +0200
@@ -25,6 +25,8 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
+#include <unistd.h>
+#include <stddef.h>
 #include <SFML/Graphics/Image.hpp>
 #include <SFML/Graphics/ImageLoader.hpp>
 #include <SFML/Graphics/RenderWindow.hpp>
diff -Naur SFML-1.6.orig/src/SFML/Graphics/PostFX.cpp SFML-1.6/src/SFML/Graphics/PostFX.cpp
--- SFML-1.6.orig/src/SFML/Graphics/PostFX.cpp	2010-01-27 15:00:05.000000000 +0100
+++ SFML-1.6/src/SFML/Graphics/PostFX.cpp	2012-09-23 19:25:25.505300968 +0200
@@ -26,6 +26,8 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
+#include <unistd.h>
+#include <stddef.h>
 #include <SFML/Graphics/PostFX.hpp>
 #include <SFML/Graphics/RenderWindow.hpp>
 #include <SFML/Graphics/GraphicsContext.hpp>
diff -Naur SFML-1.6.orig/src/SFML/Graphics/RenderTarget.cpp SFML-1.6/src/SFML/Graphics/RenderTarget.cpp
--- SFML-1.6.orig/src/SFML/Graphics/RenderTarget.cpp	2010-01-27 15:00:05.000000000 +0100
+++ SFML-1.6/src/SFML/Graphics/RenderTarget.cpp	2012-09-23 19:24:35.953300966 +0200
@@ -25,6 +25,8 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
+#include <unistd.h>
+#include <stddef.h>
 #include <SFML/Graphics/RenderTarget.hpp>
 #include <SFML/Graphics/Drawable.hpp>
 #include <SFML/Graphics/GraphicsContext.hpp>
diff -Naur SFML-1.6.orig/src/SFML/Graphics/RenderWindow.cpp SFML-1.6/src/SFML/Graphics/RenderWindow.cpp
--- SFML-1.6.orig/src/SFML/Graphics/RenderWindow.cpp	2010-01-27 15:00:05.000000000 +0100
+++ SFML-1.6/src/SFML/Graphics/RenderWindow.cpp	2012-09-23 19:28:40.236300966 +0200
@@ -25,6 +25,8 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
+#include <unistd.h>
+#include <stddef.h>
 #include <SFML/Graphics/RenderWindow.hpp>
 #include <SFML/Graphics/Drawable.hpp>
 #include <SFML/Graphics/Image.hpp>
diff -Naur SFML-1.6.orig/src/SFML/Graphics/Sprite.cpp SFML-1.6/src/SFML/Graphics/Sprite.cpp
--- SFML-1.6.orig/src/SFML/Graphics/Sprite.cpp	2010-01-27 15:00:05.000000000 +0100
+++ SFML-1.6/src/SFML/Graphics/Sprite.cpp	2012-09-23 19:27:41.924300968 +0200
@@ -25,6 +25,8 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
+#include <unistd.h>
+#include <stddef.h>
 #include <SFML/Graphics/Sprite.hpp>
 #include <SFML/Graphics/Image.hpp>
 #include <SFML/Graphics/GraphicsContext.hpp>
diff -Naur SFML-1.6.orig/src/SFML/Graphics/String.cpp SFML-1.6/src/SFML/Graphics/String.cpp
--- SFML-1.6.orig/src/SFML/Graphics/String.cpp	2010-01-27 15:00:05.000000000 +0100
+++ SFML-1.6/src/SFML/Graphics/String.cpp	2012-09-23 19:31:08.332300966 +0200
@@ -25,6 +25,8 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
+#include <unistd.h>
+#include <stddef.h>
 #include <SFML/Graphics/String.hpp>
 #include <SFML/Graphics/Image.hpp>
 #include <SFML/Graphics/GraphicsContext.hpp>
diff -Naur SFML-1.6.orig/src/SFML/Window/Linux/Joystick.cpp SFML-1.6/src/SFML/Window/Linux/Joystick.cpp
--- SFML-1.6.orig/src/SFML/Window/Linux/Joystick.cpp	2010-01-27 15:00:27.000000000 +0100
+++ SFML-1.6/src/SFML/Window/Linux/Joystick.cpp	2012-09-23 19:12:09.461300157 +0200
@@ -27,6 +27,7 @@
 ////////////////////////////////////////////////////////////
 #include <SFML/Window/Joystick.hpp>
 #include <sstream>
+#include <unistd.h>
 
 
 namespace sf
diff --git a/src/SFML/Makefile b/src/SFML/Makefile
index d8f633e..6812c83 100755
--- a/src/SFML/Makefile
+++ b/src/SFML/Makefile
@@ -17,9 +17,9 @@ endif
 
 export CC         = gcc
 export CPP        = g++
-export CFLAGS     = -W -Wall -pedantic -I../../../include -I../../ $(DEBUGFLAGS) $(BUILDFLAGS)
-export CFLAGSEXT  = -I../../../include -I../.. $(DEBUGFLAGS) $(BUILDFLAGS)
-export LDFLAGS    = $(LINKFLAGS)
+export CFLAGS     = -W -Wall -pedantic -I../../../include -I../../ $(DEBUGFLAGS) $(BUILDFLAGS) -I$(HOME)/$(ARCH)/include -I$(HOME)/$(ARCH)/include/freetype
+export CFLAGSEXT  = -I../../../include -I../.. $(DEBUGFLAGS) $(BUILDFLAGS) 
+export LDFLAGS    = $(LINKFLAGS) -L$(HOME)/$(ARCH)/lib
 export LIBPATH    = ../../../lib
 export VERSION    = 1.6
 export CP         = cp
diff --git a/src/SFML/Makefile b/src/SFML/Makefile
index d8f633e..0d0e18b 100755
--- a/src/SFML/Makefile
+++ b/src/SFML/Makefile
@@ -27,7 +27,7 @@ export LN         = ln
 export LNFLAGS    = -s -f
 export AR         = ar
 export ARFLAGS    = rcs
-export DESTDIR    = /usr/local
+export DESTDIR    = $(MYPREFIX)
 export DESTLIBDIR = $(DESTDIR)/lib
 export DESTINCDIR = $(DESTDIR)/include
 
diff --git a/lib/libsfml-audio.so.1.6 b/lib/libsfml-audio.so.1.6
index 09d117d..5c5ec01 100755
Binary files a/lib/libsfml-audio.so.1.6 and b/lib/libsfml-audio.so.1.6 differ
diff --git a/lib/libsfml-graphics.so.1.6 b/lib/libsfml-graphics.so.1.6
index 40f2644..d72fdeb 100755
Binary files a/lib/libsfml-graphics.so.1.6 and b/lib/libsfml-graphics.so.1.6 differ
diff --git a/src/SFML/Graphics/Makefile b/src/SFML/Graphics/Makefile
index 4de224d..bb0a33a 100755
--- a/src/SFML/Graphics/Makefile
+++ b/src/SFML/Graphics/Makefile
@@ -33,7 +33,7 @@ $(OBJ): %.o: %.cpp
 	$(CPP) -o $@ -c $< $(CFLAGS) -I/usr/include/freetype2
 
 $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB): %.o: %.c
-	$(CC) -o $@ -c $< $(CFLAGSEXT) -DSTBI_FAILURE_USERMSG
+	$(CC) -o $@ -c $< $(CFLAGSEXT) -DSTBI_FAILURE_USERMSG $(CFLAGS)
 
 .PHONY: clean mrproper
 
